src/backend/tty.rs

```diff
@@ -1707,7 +1738,7 @@ impl Tty {
                 };
 
                 let change_mode = surface.compositor.pending_mode() != mode;
-                let change_vrr = surface.vrr_enabled != config.variable_refresh_rate;
+                let change_vrr = surface.vrr_enabled != config.variable_refresh_rate.always_on();
```

Comment:
This seems like it should check whether some surface is enabling VRR currently on this output.

Reply:
Mhmm I am not sure the best way to do that. This just makes it false until the next refresh.

Reply:
Maybe something like, put a bool into OutputState that indicates whether, to the best of our knowledge, VRR should be enabled? Update it in `refresh_on_demand_vrr()`.

Reply:
The problem is that `refresh_on_demand_vrr` which is called in redraw does not occur before this function, so this won't do any good. I still do not see what the issue is with how it is now, where it just defaults to false. If there is an enabling surface visible, then the display will to VRR upon next redraw anyway.

Reply:
I'm just thinking of a case like changing the output scale or position while there's a surface forcing VRR. Currently this will also toggle the VRR KMS property off and on. Not a big deal I guess, but also when this happens, the output state is not erased, so if you put the last frame's VRR flag there, it should work?

Reply:
Uh yes I forgot about that case. I was thinking when switching from always on to on demand. I will add a flag there in the output state.

Reply:
I don't think a simple VRR flag is enough. The window rule config could have changed at the same time, and so what enabled the VRR previously may not apply anymore. There are also edge cases with the other transitions. Overall, I don't think it is worth the trouble to implement it.